qiskit/quantum_info/operators/symplectic/pauli.py

```diff
@@ -0,0 +1,1027 @@
+
+    def __len__(self):
+        """Return the number of qubits in the Pauli."""
+        return self.num_qubits
+
+    def __getitem__(self, qubits):
+        """Return the unsigned Pauli group Pauli for subset of qubits."""
+        # Set group phase to 0 so returned Pauli is always +1 coeff
+        if isinstance(qubits, (int, np.int)):
+            qubits = [qubits]
+        return Pauli(self.z[qubits], self.x[qubits])
+
+    def __setitem__(self, qubits, value):
+        """Update the Pauli for a subset of qubits."""
+        if not isinstance(value, Pauli):
+            value = Pauli(value)
+        self._z[0, qubits] = value.z
+        self._x[0, qubits] = value.x
+        # Add extra phase from new Pauli to current
+        self._phase += value._phase
+
+    def delete(self, qubits):
+        """Return a Pauli with qubits deleted.
+
+        Args:
+            qubits (int or list): qubits to delete from Pauli.
+
+        Returns:
+            Pauli: the resulting Pauli with the specified qubits removed.
+
+        Raises:
+            QiskitError: if ind is out of bounds for the array size or
+                         number of qubits.
+        """
+        if isinstance(qubits, (int, np.int)):
+            qubits = [qubits]
+        if max(qubits) > self.num_qubits - 1:
+            raise QiskitError(
+                "Qubit index is larger than the number of qubits "
+                "({}>{}).".format(max(qubits), self.num_qubits - 1))
+        if len(qubits) == self.num_qubits:
+            raise QiskitError("Cannot delete all qubits of Pauli")
+        z = np.delete(self._z, qubits, axis=1)
+        x = np.delete(self._x, qubits, axis=1)
+        return Pauli(z, x, phase=self.phase)
+
+    def insert(self, qubits, value):
+        """Insert a Pauli at specific qubit value.
+
+        Args:
+            qubits (int or list): qubits index to insert at.
+            value (Pauli): value to insert.
+
+        Returns:
+            Pauli: the resulting Pauli with the entries inserted.
+
+        Raises:
+            QiskitError: if the insertion qubits are invalid.
+        """
+        if not isinstance(value, Pauli):
+            value = Pauli(value)
+
+        # Initialize empty operator
+        ret_qubits = self.num_qubits + value.num_qubits
+        ret = Pauli(np.zeros(ret_qubits, dtype=np.bool),
+                    np.zeros(ret_qubits, dtype=np.bool))
+        if isinstance(qubits, (int, np.int)):
+            if value.num_qubits == 1:
+                qubits = [qubits]
+            else:
+                qubits = list(range(qubits, qubits + value.num_qubits))
+        if len(qubits) != value.num_qubits:
+            raise QiskitError(
+                "Number of indices does not match number of qubits for "
+                "the inserted Pauli ({}!={})".format(len(qubits),
+                                                     value.num_qubits))
+        if max(qubits) > ret.num_qubits - 1:
```

Comment:
It's good to raise an error if `min(qubits) < 0` too.

Reply:
I recommend to add the following check as well as `if max(qubits) > ret.num_qubits - 1`.
```
        if min(qubits) < 0:
            raise QiskitError(
                "Index must be non-negative for combined Pauli number of qubits "
                "({}<0).".format(min(qubits))
````